third_party/amd/lib/TritonAMDGPUToLLVM/BuiltinFuncToLLVM.cpp

```diff
@@ -98,6 +109,45 @@ class CallOpConversion : public mlir::RewritePattern {
     rewriter.replaceOp(callOp, loadVal);
     return mlir::success();
   }
+
+  LogicalResult convertToLLVMIntrinsic(LLVM::CallOp callOp,
+                                       mlir::PatternRewriter &rewriter) const {
+    StringRef calleeName = callOp.getCallee().value();
+
+    auto operands = callOp.getOperands();
+    auto result = callOp.getResult();
+
+    LLVM::LLVMFunctionType calleeType = callOp.getCalleeType().value();
+    Type returnType = calleeType.getReturnType();
+
+    LLVM::FastmathFlagsAttr flags;
+    auto loc = callOp.getLoc();
+
+    Operation *replacementOp = nullptr;
+    if (calleeName == "__triton_hip_iabs") {
```

Comment:
Do you expect iabs, fabs, and llrint are the only cases that need special care? For now it seems ok. But in the future, we might have more cases coming from
- more math functions
- for some math functions, we are not happy with the impl in ocml_xxx

Reply:
> Do you expect iabs, fabs, and llrint are the only cases that need special care?

No, there will definitely a handful of extra functions which would require a similar treatment. Yes, it may look like a not a scalable solution but I believe it is the best what we can do. One can consider these if-statements like a rewrite table.

Btw, @zhanglx13 and @antiagainst, the current implementation cannot handle the AOT compilation. I am working on a patch which is going to fix it.